tests/Earthfile

```diff
@@ -1128,6 +1129,23 @@ help:
     ENV EARTHLY_SHOW_HIDDEN=1
     RUN earthly --help | grep 'buildkit-volume-name'
 
+doc:
+    DO +RUN_EARTHLY --earthfile=target-docs.earth --extra_args="doc" --target="" --output_contains="TARGETS:
+  +documented-target
+    documented-target is a target with documentation
+    that spans multiple lines.
+
+    It also has a separator between paragraphs."
+    DO +RUN_EARTHLY --earthfile=target-docs.earth --extra_args="doc" --target="" --grep_flags="-v" --output_contains="undocumented-target"
+    DO +RUN_EARTHLY --earthfile=target-docs.earth --extra_args="doc" --target="" --grep_flags="-v" --output_contains="incorrectly-documented-target"
+    DO +RUN_EARTHLY --earthfile=target-docs.earth --extra_args="doc" --target="+documented-target" --output_contains="+documented-target
+  documented-target is a target with documentation
+  that spans multiple lines.
+
+  It also has a separator between paragraphs."
+    DO +RUN_EARTHLY --earthfile=target-docs.earth --extra_args="doc" --target="+undocumented-target" --output_contains="No doc comment found"
+    DO +RUN_EARTHLY --earthfile=target-docs.earth --extra_args="doc" --target="+incorrectly-documented-target" --output_contains="No doc comment found"
```

Comment:
should this cause earthly to exit with a non-zero exit code? there's a `--should_fail=true` flag for `DO +RUN_Earthly` that can be used.

Reply:
Good point.  I noticed that arg, but just didn't think about whether or not `doc` should have a non-zero exit code for this case - but it probably should.

Reply:
FWIW, I made `earthly doc` actually error out if a specific target is requested but no documentation exists for that target. I think it seems reasonable. Plus as an experimental sub-command, we can always soften up those errors later.